packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts

```diff
@@ -292,25 +289,53 @@ export class PureDataObjectFactory<TObj extends PureDataObject<O, S, E>, O, S, E
         runtime: IContainerRuntimeBase,
         initialState?: S,
     ): Promise<TObj> {
-        return this.createInstanceCore(runtime, [this.type], initialState);
+        return this.createNonRootInstanceCore(
+            runtime,
+            [this.type],
+            initialState);
     }
 
-    protected async createInstanceCore(
+    /**
+     * Creates a new root instance of the object. Uses container's registry to find this factory.
+     * It's expected that only container owners would use this functionality, as only such developers
+     * have knowledge of entries in container registry.
```

Comment:
It feels just a bit weird to have this comment (and I guess the `IContainerRuntime` typing below) playing primary gatekeeper to using this API when it lets you specify the data store id whereas the alternate API does not.  Hopefully no one abuses it 🙃

Reply:
I think this is the right approach: https://github.com/microsoft/FluidFramework/pull/3700
But while we wait to settle it, I'd use any other opportunity to protect API surface against misuse.
Yes, it's easy to overcome it (simple cast), but it's at least something.